src/main/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptor.java

```diff
@@ -205,26 +210,28 @@ public ListBoxModel doFillHardwareIdItems(
             @QueryParameter String hardwareId,
             @QueryParameter String endPointUrl, @QueryParameter String identity, @QueryParameter String credential, @QueryParameter String zone
     ) {
-
         ListBoxModel m = new ListBoxModel();
         m.add("None specified", "");
-
+        final String valueOrEmpty = Util.fixNull(hardwareId);
+        boolean existingValueFound = valueOrEmpty.isEmpty();
         try {
-            final Openstack openstack = Openstack.Factory.get(endPointUrl, identity, credential, zone);
-            for (Flavor flavor : openstack.getSortedFlavors()) {
-                m.add(String.format("%s (%s)", flavor.getName(), flavor.getId()), flavor.getId());
+            if (haveServerDetails(endPointUrl, identity, credential, zone)) {
+                final Openstack openstack = Openstack.Factory.get(endPointUrl, identity, credential, zone);
+                for (Flavor flavor : openstack.getSortedFlavors()) {
+                    final String value = flavor.getId();
+                    final String displayText = String.format("%s (%s)", flavor.getName(), value);
+                    m.add(displayText, value);
+                    existingValueFound |= valueOrEmpty.equals(value);
+                }
             }
-            return m;
         } catch (AuthenticationException | FormValidation | ConnectionException ex) {
             LOGGER.log(Level.FINEST, "Openstack call failed", ex);
         } catch (Exception ex) {
             LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
         }
-
-        if (Util.fixEmpty(hardwareId) != null) {
+        if (!existingValueFound) {
```

Comment:
FYI dc2e17aaf57aea1036cc801e076d61b25f80da8b is what the code looks like with `if (!hasValue(m, valueOrEmpty)) {`
We can either keep those changes or exclude them.